### ISSUE TITLE
F# compiler warnings in .NET Core 3.1

### ISSUE THREAD

**Issue by frankebersoll** [dotnet-websharper/core#1093]
## Situation
- The current `dotnet new` templates create .NET Core 3.1 projects.
- When using `[<Rpc>]` with records as return type, the compiler emits a warning:
```fsharp
module Client =
    [<JavaScript>]
    type SomeRecord = { Name : string }
    [<Rpc>] 
    let DoSomething () = async { return { Name = "Yo." } }
```
```console
WebSharper warning FS9002: Failed to load return type 'TestProjectCore31.Client+SomeRecord, TestProjectCore31' to verify encoding to JSON. at TestProjectCore31.Client.DoSomething
```
- The problem only occurs when targeting .NET Core 3.1. Compiling the same code against .NET Core 3.0 doesn't produce the warning.

## Cause
The Verifier uses `canEncodeToJson` to check the return type, which calls `ASTTypes.LoadType(Type)` and raises the following exception:
```console
Could not load file or assembly 'System.Runtime, Version=4.2.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. Could not find or load a specific file. (Exception from HRESULT: 0x80131621)
   at System.RuntimeTypeHandle.GetTypeByName(String name, Boolean throwOnError, Boolean ignoreCase, Boolean reflectionOnly, StackCrawlMarkHandle stackMark, IntPtr pPrivHostBinder, Boolean loadTypeFromPartialName, ObjectHandleOnStack type, ObjectHandleOnStack keepalive)
   at System.RuntimeTypeHandle.GetTypeByName(String name, Boolean throwOnError, Boolean ignoreCase, Boolean reflectionOnly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean loadTypeFromPartialName)
   at System.RuntimeType.GetType(String typeName, Boolean throwOnError, Boolean ignoreCase, Boolean reflectionOnly, StackCrawlMark& stackMark)
   at System.Type.GetType(String typeName, Boolean throwOnError)
   at WebSharper.Core.AST.Reflection.LoadType(Type t) in C:\Sources\github\websharper-core\src\compiler\WebSharper.Core\ASTTypes.fs:line 847
```

## Similar issues
- https://stackoverflow.com/questions/59610009/net-core-3-1-could-not-load-file-or-assembly-system-runtime-version-4-2-2-0
- https://stackoverflow.com/questions/59563311/how-to-solve-runtime-error-in-visual-studio
- https://github.com/dotnet/orleans/issues/6215

**Comment by frankebersoll** [comment]
I managed to debug WebSharper.FSharp using a response file, but I couldn't figure out how to create a  minimal working example that I could push.

**Comment by granicz** [comment]
Thanks for the detailed report!

**Comment by drssoccer55** [comment]
I ran into a very similar error where System.Runtime 4.2.2.0 cannot be loaded except with WebSharper.Core.AST.Reflection.LoadTypeDefinition. I used the websharper single page template with netcoreapp3.1 and tried to include an embedded resource in a bundle project. If I change to netcoreapp3.0 it works but that's not a workable solution for me. One thing I noticed is that my dll uses System.Runtime 4.2.2.0 but the reference to websharper.core uses 4.1.2.0 and when I switch to 3.0 they are both on 4.1.20

![image](https://user-images.githubusercontent.com/23265878/79495232-2692b100-7fd9-11ea-904c-98532a4ed0fc.png)
